### PR TITLE
Line chart: Metrices and Dimensions, Legend, Chart styles features implementation on config panel

### PR DESCRIPTION
**Pull request by rinku-kumar-psl**
### Description
[Describe what this change achieves]
The below features are implemented in the config panel in this PR:

Legend:

- Show Legend
- Position

Chart styles:

- Mode (Line, Bars, Points, Lines+Points)
- Interpolation (Linear, Smooth, Step Before, Step After)
- Line Width
- Fill Opacity
- Point Size

Metrix and dimension UI Integration on Data Configuration.

### Issues Resolved
[List any issues this PR will resolve]
opensearch-project/observability#697 opensearch-project/dashboards-observability#125

Here are the visuals and implementation of new Data configs on the Line chart as per the issue https://github.com/opensearch-project/observability/issues/697
https://github.com/opensearch-project/observability/issues/697#issuecomment-1125181489

### Check List
- [ ] New functionality includes above listed Legend and Chart styles features in the Data config panel and Metrix and Dimension Integration.
  - [ ] All tests that pass, including unit test, and integration test, except a few test cases, are failing at my end which is not related to this functionality.
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

@spattnaik

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    },
+    [handleConfigChange, vizState]
+  );
+
+  /* To update the schema options based on current style mode selection */
+  const currentSchemas = useMemo(() => {
+    if (!vizState?.style || vizState?.style === "lines") {
+      return schemas.filter((schema: IConfigPanelOptionSection) => schema.mapTo !== 'pointSize');
+    }
+    if (vizState?.style === "bar") {
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx:39]: do we need this in config line chart file?

File: dashboards-observability/public/components/visualizations/charts/lines/line.tsx [hunk 2]
```diff
@@ -37,21 +45,69 @@ export const Line = ({ visualizations, layout, config }: any) => {
...
+      };
+      const fillProperty = {
+        fill: 'tozeroy',
+        fillcolor: fillColor,
+      };
+      const multiYaxis = { yaxis: `y${index + 1}` };
+      multiMetrics = {
+        ...multiMetrics,
+        [`yaxis${index + 1}`]: {
+          // title: `yaxis${index + 1} title`,
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/lines/line.tsx]: remove commented code

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx (hunk 1 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_line_chart_styles.tsx:39]: Hi @abasatwar, As we can switch Line traces between BAR and LIne, So this check is there.

File: dashboards-observability/public/components/visualizations/charts/lines/line.tsx (hunk 2 above)
Comment (rinku-kumar-psl) [reply dashboards-observability/public/components/visualizations/charts/lines/line.tsx]: Thanks!
Added TODO comment.

**Review by abasatwar — APPROVED** [review]
looks good.

**Pull request merged**